src/wasm-interpreter.h

```diff
@@ -170,11 +178,71 @@ class ExpressionRunner : public OverriddenVisitor<SubType, Flow> {
   }
 
 public:
-  ExpressionRunner(Index maxDepth) : maxDepth(maxDepth) {}
+  enum FlagValues {
+    // By default, just evaluate the expression, i.e. all we want to know is
+    // whether it computes down to a concrete value, where it is not necessary
+    // to preserve side effects like those of a `local.tee`.
+    DEFAULT = 0,
+    // Be very careful to preserve any side effects, like those of a
+    // `local.tee`, for example when we are going to replace the expression
+    // afterwards.
+    PRESERVE_SIDEEFFECTS = 1 << 0,
+    // Traverse through function calls, attempting to compute their concrete
+    // value. Must not be used in function-parallel scenarios, where the called
+    // function might or might not have been optimized already to something we
+    // can traverse successfully, in turn leading to non-deterministic behavior.
+    TRAVERSE_CALLS = 1 << 1
+  };
+
+  // Flags indicating special requirements, for example whether we are just
+  // evaluating (default), also going to replace the expression afterwards or
+  // executing in a function-parallel scenario. See FlagValues.
+  typedef uint32_t Flags;
+
+  Flags flags;
+
+  ExpressionRunner(Flags flags = FlagValues::DEFAULT) : flags(flags) {}
```

Comment:
You can add another constructor that accepts `module, maxDepth, maxLoopIteractions` and sets them, and then forward to that constructor from the child. Then the child looks like `ParentConstructor(param x, param y) : ChildConstructor(x, y)`